iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -245,6 +245,53 @@ def LinalgExt_FftOp : LinalgExt_Op<"fft", [
   }];
 }
 
+def LinalgExt_ReverseOp : LinalgExt_Op<"reverse", [
+  DeclareOpInterfaceMethods<TiledOpInterface,
+                            [
+                              "generateScalarImplementation"
+                            ]>,
+  DeclareOpInterfaceMethods<LinalgExtInterface,
+                            // ReverseOp does not have a region, so we have to
+                            // overwrite the method.
+                            ["payloadUsesValueFromOperand"]>
+]> {
+  let summary = "Reverse operator";
+  let description = [{
+    A temporary solution of a reverse op. The loop bound of the reverse
+    dimension is half of the shape because we can simply swap elements. E.g.,
+
+    for (int i = 0; i < n / 2; ++i) {
```

Comment:
Wondering if this will have any implications on tile + distribute, but lets keep it this way.

Reply:
I think it does not affect tile + distribute. We can either iterate the whole tensor or half of the tensor. The main difference is whether we can reverse the operand in place. I.e., have one buffer for the op or two buffers for the op. With this approach, we can use a single buffer.